src/pcre2_compile.c

```diff
@@ -3737,9 +3716,12 @@ while (ptr < ptrend)
         have a warning mode, so we give an error, because this is likely an
         error on the user's part. */
 
-        if (ptr < ptrend - 1 && *ptr == CHAR_MINUS &&
-            ptr[1] != CHAR_RIGHT_SQUARE_BRACKET)
+        if (ptr < ptrend && *ptr == CHAR_MINUS &&
+            !((options & PCRE2_ALT_EXTENDED_CLASS) != 0 && ptr + 1 < ptrend &&
+              ptr[1] == CHAR_MINUS) &&
+            (ptr + 1 == ptrend || ptr[1] != CHAR_RIGHT_SQUARE_BRACKET))
           {
+          // TODO: We need a better lookahead here to match Perl; should skip \Q\E, and whitespace in /xx mode.
```

Comment:
Maybe a `/* TODO: EC revert */` will be a better prefix. Then we can just find `TODO: EC` to update all related things.

Note: PCRE2 does not use // comments at all.

Reply:
That's because C90 did not have // comments, but nowadays I do occasionally use them for "temporary" comments that can easily be found for removal.

Reply:
I have added TODOs as indicated by Zoltan, plus some more all.

All of them have the string `TODO: [EC]` and a link to the GitHub issue.